plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java

```diff
@@ -143,9 +143,11 @@ public CompilerResult performCompile( CompilerConfiguration config )
 
         if ( ( getLogger() != null ) && getLogger().isInfoEnabled() )
         {
+            String to = ( config.getWorkingDirectory() == null ) ? config.getOutputLocation() :
+                config.getWorkingDirectory().toPath().relativize( destinationDir.toPath() ).toString();
             getLogger().info( "Compiling " + sourceFiles.length + " " +
                                   "source file" + ( sourceFiles.length == 1 ? "" : "s" ) +
-                                  " to " + destinationDir.getAbsolutePath() );
+                                  " to " + to );
```

Comment:
Could these duplications be avoided?

Reply:
> Could these duplications be avoided?

?? not sure to understand your concern?

Reply:
you mean create a method in AbstractCompiler for logging, and have plexus compilers use this method?

https://codehaus-plexus.github.io/plexus-compiler/plexus-compiler-api/apidocs/org/codehaus/plexus/compiler/AbstractCompiler.html

Reply:
`to` and message logged/being output are calculated twice (Javac, AspecJCompiler) in the same way.

Reply:
good idea, done in 37f6843f4dfbf0a8a9caa185926d16ab8b55f0e8

Reply:
Great!